### PR TITLE
DOC: Note FFT type promotion

### LINKED ISSUE #14892: Numpy FFTs return data types don't match input data types
Unlike other functions, the NumPy FFT functions do not return arrays whose data types match the input arrays. Instead, `complex64` is promoted to `complex128`.

### Reproducing code example:

```python
import sys
import numpy as np

print(np.__version__, sys.version)

input = np.ones((5, 5), dtype='complex64')

output = np.square(input)
assert output.dtype == 'complex64', f"{output.dtype} square"  # assert passes

output = np.fft.fft2(input)
assert output.dtype == 'complex64', f"{output.dtype} fft"  # assert fails
```

### Numpy/Python version information:

1.17.3 3.8.0 | packaged by conda-forge | (default, Nov  6 2019, 23:20:36) 
[GCC 7.3.0]

### PR DESCRIPTION
**Pull request by carterbox**
The NumPy FFT implementation will promote `float32` types to `float64` types. This may not be desired for some applications, but the SciPy implementation supports more data types.

Closes #14892

### CODE REVIEW

File: numpy/fft/__init__.py [hunk 1]
```diff
@@ -118,8 +118,16 @@
...
+Type Promotion
+--------------
+
+`numpy.fft` is inconsistent with other NumPy functions as it promotes
+`float32` and `complex64` arrays to `float64` and `complex128` arrays.
+For an FFT implementation that does not promote input arrays, see `scipy.fftpack`.
+
 Normalization
 -------------
+
```
Comment (mattip) [inline numpy/fft/__init__.py]: why this change?

File: numpy/fft/__init__.py [hunk 2]
```diff
@@ -118,8 +118,16 @@
 It differs from the forward transform by the sign of the exponential
 argument and the default normalization by :math:`1/n`.
 
+Type Promotion
+--------------
+
+`numpy.fft` is inconsistent with other NumPy functions as it promotes
+`float32` and `complex64` arrays to `float64` and `complex128` arrays.
+For an FFT implementation that does not promote input arrays, see `scipy.fftpack`.
+
```
Comment (mattip) [inline numpy/fft/__init__.py]: I think 
```
"`numpy.fft` promotes ``float32`` and ``complex64`` arrays to ``float64`` and
``complex128`` arrays respectively."
```

 reads better. Note also the double backticks needed for the dtypes.

File: numpy/fft/__init__.py (hunk 1 above)
Comment (carterbox) [reply numpy/fft/__init__.py]: To make this heading consistent with the surrounding headings. It appears that all the other headings have white space both above and below. This heading only had whitespace above.

**Comment by mattip** [comment]
LGTM, [here](https://10178-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/routines.fft.html?highlight=fft#type-promotion) is the new section as rendered in HTML.

**Review by mattip — APPROVED** [review]

**Comment by carterbox** [comment]
Thanks for the HTML link @mattip.

**Comment by mattip** [comment]
Thanks @carterbox
